### DISCUSSION TITLE
Is "ref" a reserved word?

### DISCUSSION THREAD

**Discussion by oltodo** [react-hook-form/react-hook-form#7441] [Q&A]
Hello,

I'm trying to move from version 7.15.4 to 7.12.5 and I observe a strange behavior. Actually, it seems that using the term `ref` as a form's field prevents the `isDirty` state value from being properly updated. When I change it for another term, it works fine.

Sorry because I've not been able to reproduce this issue in a codesanbox instance. What I can say is that my form is quite complex. It looks something like:

```js
{
  id: '...',
  blocks: [
    {
      type: 'foo',
      data: {
        items: [{
          ref: '...', // <- the issue
          title: '...'
        }]
      }
    },
    {
      type: 'bar',
      data: {
        // other kind of data
      }
    },
    {
      type: 'foo',
      data: {
        // ...
      }
    }
    // and so on...
  ]
}
```

Also I use Yup as form validator.

I realize it could be a little bit tricky to investigate basing on the above description. You can take a look on the [repo](https://github.com/EgliseLyonGerland/LiturgyMaker) I've been working on. Thanks 😉 .

**Comment by bluebill1049** [comment]
yes, `ref` is a reserved word for input reference.

**Reply by davbrito** [reply]
There should be a runtime warning when using "ref" as a field name. And a mention on the docs.

**Comment by man-trackunit** [comment]
For me `ref` makes sense to be reserved but `type` also seems reserved which haunts me a bit since it's one of the top 5 most important property keys on pur data model 🙈
